File: backend/semgrep_rules/tests/java/kisa_sw_34_toctou.java
import java.nio.file.*;
import java.io.*;
class kisa_sw_34_toctou {
    InputStream unsafe(Path path) throws Exception {
        if (Files.exists(path)) {
            // ruleid: kisa.sw34.java.check-then-use-file
            return Files.newInputStream(path);
        }
        return null;
    }
    OutputStream safe(Path path) throws Exception {
        // ok: kisa.sw34.java.check-then-use-file
        return Files.newOutputStream(path, StandardOpenOption.CREATE_NEW);
    }
}

File: backend/semgrep_rules/tests/python/kisa_sw_34_toctou.py
import os
def unsafe(path):
    if os.path.exists(path):
        # ruleid: kisa.sw34.python.check-then-use-file
        return open(path, "r")
def safe(path):
    # ok: kisa.sw34.python.check-then-use-file
    try:
        return open(path, "x")
    except FileExistsError:
        return None
